cleartk-berkeleyparser/src/main/java/org/cleartk/berkeleyparser/DefaultBerkeleyTokenizer.java

```diff
@@ -0,0 +1,130 @@
+import org.apache.uima.jcas.JCas;
+import org.apache.uima.jcas.cas.FSArray;
+import org.apache.uima.jcas.tcas.Annotation;
+import org.apache.uima.resource.ResourceInitializationException;
+import org.cleartk.syntax.constituent.type.TerminalTreebankNode;
+import org.cleartk.syntax.constituent.type.TopTreebankNode;
+import org.cleartk.token.type.Token;
+
+import edu.berkeley.nlp.io.LineLexer;
+
+/**
+ * <br>
+ * Copyright (c) 2010, Regents of the University of Colorado <br>
+ * All rights reserved.
+ * 
+ * @author Majid Laali
+ */
+public class DefaultBerkeleyTokenizer<TOKEN_TYPE extends Annotation, SENTENCE_TYPE extends Annotation> extends JCasAnnotator_ImplBase {
+  public static final String PARAM_INPUT_TYPES_HELPER_CLASS_NAME = "inputTypesHelperClassName";
+
+  public static AnalysisEngineDescription getDescription() throws ResourceInitializationException{
+    return AnalysisEngineFactory.createEngineDescription(DefaultBerkeleyTokenizer.class, 
+        PARAM_INPUT_TYPES_HELPER_CLASS_NAME, DefaultInputTypesHelper.class.getName());
+  }
+
+  @ConfigurationParameter(
+      name = PARAM_INPUT_TYPES_HELPER_CLASS_NAME,
+      defaultValue = "org.cleartk.berkeleyparser.DefaultInputTypesHelper",
+      mandatory = true)
+  protected String inputTypesHelperClassName;
+
+  private InputTypesHelper<TOKEN_TYPE, SENTENCE_TYPE> inputTypesHelper;
+  private LineLexer tokenizer = new LineLexer();
+  
+  @SuppressWarnings("unchecked")
+  @Override
+  public void initialize(UimaContext ctx) throws ResourceInitializationException {
+    super.initialize(ctx);
+
+    inputTypesHelper = InitializableFactory.create(
+        ctx,
+        inputTypesHelperClassName,
+        InputTypesHelper.class);
+  }
+  
+  public List<TOKEN_TYPE> tokenize(SENTENCE_TYPE sent) {
+    try {
+      int base = sent.getBegin();
+      String strSent = sent.getCoveredText();
+      JCas jCas = sent.getCAS().getJCas();
+      List<String> strTokens = tokenizer.tokenizeLine(strSent);
+      List<TOKEN_TYPE> tokens = new ArrayList<>();
+      
+      int index = 0;
+      for (String strToken: strTokens){
+        int nextIndex = strSent.indexOf(strToken, index);
+        if (nextIndex == -1){
+          System.err.print(String.format("Cannot find token <%s> in the sentence <%s>: \n", strToken, strSent));
+          continue;
+        }
+        index = nextIndex;
+        
+        int begin = base + index;
+        int end = begin + strToken.length();
+        TOKEN_TYPE token = inputTypesHelper.buildToken(jCas, begin, end);
+        if (token != null){
+          token.addToIndexes();
+          tokens.add(token);
+        }
+        index += strToken.length();
+      }
+      return tokens;
+    } catch (IOException e) {
+      e.printStackTrace();
+    } catch (CASException e) {
+      e.printStackTrace();
+    }
+    return null;
+    
+  }
+
+  public void setPosTags(List<Token> tokens, TopTreebankNode topTreebankNode) {
```

Comment:
It seems odd for a tokenizer to 1) set POS tags and 2) require knowledge of TreebankNodes. It also appears this method is not used anywhere within the DefaultBerkeleyTokenizer nor anywhere else.

I would remove this method from both this and the tokenizer interface.